x-pack/legacy/plugins/siem/public/components/ml/api/parse_error_message.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const parseError = async (response: Response): Promise<string | null> => {
+  if (!response.ok) {
+    const body = await parseJsonFromBody(response);
+    if (body != null && body.message) {
+      return body.message;
+    } else {
+      return `Network Error: ${response.statusText}`;
```

Comment:
Question, why can we just throw an error and have all the logic of toaster in the `catch`

Reply:
I think so, I will update the code and add tests around this area. Thanks for the feedback.

Reply:
Updated and added unit tests around this part of the code now.